toolkit/tools/validatechroot/validatechroot.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+	"microsoft.com/pkggen/internal/exe"
+	"microsoft.com/pkggen/internal/file"
+	"microsoft.com/pkggen/internal/logger"
+	"microsoft.com/pkggen/internal/safechroot"
+	"microsoft.com/pkggen/internal/shell"
+)
+
+var (
+	app = kingpin.New("validatechroot", "A tool to validate that the worker chroot is well configured and all dependencies are satisfied.")
+
+	toolchainRpmsDir = app.Flag("rpm-dir", "Directory that contains already built toolchain RPMs. Should contain top level directories for architecture.").Required().ExistingDir()
+	tmpDir           = app.Flag("tmp-dir", "Temporary chroot directory.").String()
+
+	workerTar      = app.Flag("worker-chroot", "Full path to worker_chroot.tar.gz").Required().ExistingFile()
+	workerManifest = app.Flag("worker-manifest", "Full path to the worker manifest file").Required().ExistingFile()
+
+	logFile  = exe.LogFileFlag(app)
+	logLevel = exe.LogLevelFlag(app)
+
+	leaveChrootFilesOnDisk = false
+)
+
+func main() {
+	app.Version(exe.ToolkitVersion)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	logger.InitBestEffort(*logFile, *logLevel)
+
+	err := validateWorker()
+
+	if err != nil {
+		logger.Log.Fatalf("Failed to validate worker. Error: %s", err)
+	}
+}
+
+func validateWorker() (err error) {
```

Comment:
nit: In general with the tools we avoid dereferencing the kingpin arguments outside of `main`. Ideally we pass all the dereferenced arguments to the function as needed so the function. This gives the benefit of making the function itself more portable across our code base and have clear inputs & outputs for the reader.